go-controller/pkg/ovn/acl.go

```diff
@@ -2,12 +2,69 @@ package ovn
 
 import (
 	"fmt"
+	"strings"
+
 	libovsdbclient "github.com/ovn-org/libovsdb/client"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+
+	knet "k8s.io/api/networking/v1"
 )
 
+func joinACLName(substrings ...string) string {
+	return strings.Join(substrings, "_")
+}
+
+// aclType defines when ACLs will be applied (direction and pipeline stage).
+// All acls of the same type will be sorted by priority, priorities for different types are independent.
+type aclType string
+
+const (
+	// lportIngress will be converted to direction="to-lport" ACL
+	lportIngress aclType = "Ingress"
+	// lportEgressAfterLB will be coverted to direction="from-lport", options={"apply-after-lb": "true"} ACL
+	lportEgressAfterLB aclType = "Egress"
+)
+
+func aclTypeToPolicyType(aclT aclType) knet.PolicyType {
+	if aclT == lportEgressAfterLB {
+		return knet.PolicyTypeEgress
+	} else {
+		return knet.PolicyTypeIngress
+	}
+}
+
+// BuildACL should be used to build ACL instead of directly calling libovsdbops.BuildACL.
+// It can properly set and reset log settings for ACL based on ACLLoggingLevels
+func BuildACL(aclName string, priority int, match, action string,
+	logLevels *ACLLoggingLevels, aclT aclType, externalIDs map[string]string) *nbdb.ACL {
+	var options map[string]string
+	var direction string
+	if aclT == lportIngress {
```

Comment:
nit: switch and panic default case might be safer here?

Reply:
Probably good to have aclType<From|To><Direction|PolicyType> methods with this consideration

Reply:
I am not sure about direction, because it is (and should be) only used in syncFunction and not it a normal workflow. it is more hacky than ok to do that, wdyt?

Reply:
If it were me I would just have it and not concern much about it, but I am ok with what you decide.
I do have the feeling that a switch with a default case would be appropriate there anyway. I know it practically cannot happen, but theoretically it could and everyone that reads this code in the future might be triggered to think about it while the switch default case panic gives a more warmy feeling to the eyes :P

Reply:
@jcaamano yeah it can also be useful if we add a new acl type at some point, updated